src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -176,7 +176,6 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
         break;
       case RATELIMIT:
         if (response.isUserOrDeviceExceedQuote(userId, deviceId)) {
```

Comment:
We should retry differently based on the response. See [Kotlin implementation](https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/utilities/InMemoryResponseHandler.kt#L66-L71) here and [comments in Slack](https://amplitude.slack.com/archives/C02E1H87E2C/p1687465582561709?thread_ts=1687465015.015709&cid=C02E1H87E2C)

https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/utilities/InMemoryResponseHandler.kt#L66-L71

Reply:
We don’t have a `eventsToRetryLater` logic available. Can we just delete `shouldReduceEventCount` and retry if don't exceed quote?

Reply:
I think [Brian's suggestion seems good](https://amplitude.slack.com/archives/C02E1H87E2C/p1687469723892789?thread_ts=1687465015.015709&cid=C02E1H87E2C). We would still need to test.

> maybe can we just add back the dropped events to the `idToBuffer` map if we drop them due to a 429?

Reply:
I dont think adding back to the idToBuffer map will actually help since that buffer is not tied to the actual buffer in the client.